### ISSUE TITLE
use lxsession or an other one with spectrwm

### ISSUE THREAD

**Issue by jerome-diver** [conformal/spectrwm#273]
which session manager can be used with spectrwm and how to, please ?

**Comment by mschilli87** [comment]
I have an Ubuntu 16.04 machine with spectrwm installed via `apt-get install`. With the default setup, I can choose spectwrm from LightDM. Maybe [the corresponding package](https://launchpad.net/ubuntu/+source/spectrwm/1.0.0-1) is helpful for you.

**Comment by jerome-diver** [comment]
@mschilli87  LightDM is not a session manager.
It is a Diesktop Manager (DM).
A session window manager can manage session of applications opened on last closed time or any named and saved sessions of the Desktop/Window Manager.
For example, XFCE and Gnome have there own session window manager.
lxsession is the one for LXDE and it maybe possible to use it with any other Desktop, but i'm not sure really (it is what i read from there own github repo). That is why i ask and i think many other users should be happy to have a session manager to no more have to open again all the last applications opened at last disconnection.
There is also linux-window-session-manager (who has its own github repo and installable with npm, the nodejs package manager), but again... may it works with spectrwm and may it work also for ALL applications GUI/TUI ?

**Comment by marcopeereboom** [comment]
I have no idea what you are asking.

**Comment by jerome-diver** [comment]
But you understand the question (i ask because maybe my english is not good) ?

**Comment by marcopeereboom** [comment]
I literally have no idea what you are after.

**Comment by jerome-diver** [comment]
Do you have an idea about what is a session manager for a desktop ?

**Comment by jerome-diver** [comment]
Ok, i suppose you are interested to have an idea about what i'm talking about... so let me try again to give me an other one chance to explain by examples:

1- i open a session (login my user to spectrwm).
2- i open pycharm
3- i move a pycharm's window (let said: the git commit one) to an other one screen (on right)
4- i open terminal and tmux through on next workspace
5-  i open hexchat on an other one workspace
6- i open a web browser in an other one workspace
7- after to work, i save the current session, and i close the session (logout), then shutdown
8- i reboot then login the same user session than step number one
9- my default session (the one saved at step 7) re-open all the last application at the same position on each workspace/screens.

that's it.
Is it more clear ?

**Comment by marcopeereboom** [comment]
Ah I see. Spectrwm is a window manager and not a session manager. I have wanted a graphic tmux myself for a long time unfortunately I am unaware of one existing. I have not heard of lwsm before.

@LordReg is this something we can "integrate" with?

**Comment by jerome-diver** [comment]
@marcopeereboom sure, spectrwm is not a window session manager, i use spectrwm and then i know it is a windows tilling manager. A window session manager works with a windows manager, that is why i ask which one can work with spectrwm or if there is a plan to build one or adapt an existing one.
I also don't want any graphic tmux, but tmux example is for show that open a terminal at session recover time (if one was opened) is fine, but to be able to recover also at tmux session is also a very good feature for a window session manager.
thank you very much for your help. Now that you know what is a window session manager you should maybe also want to have one.

**Comment by marcopeereboom** [comment]
@LordReg ?

**Comment by jerome-diver** [comment]
ok, i think this application is no more maintained actualy. Let me know when it will be up, i will install and try it again.
